frontend/routes/components/topMenu/components/visualSettingsDialog/visualSettingsDialog.component.tsx

```diff
@@ -0,0 +1,215 @@
+/**
+ *  Copyright (C) 2019 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Button, InputAdornment, List, MenuItem, Select, Switch } from '@material-ui/core';
+import { Field, Form, Formik } from 'formik';
+import * as React from 'react';
+import * as Yup from 'yup';
+import { schema } from '../../../../../services/validation';
+import { DialogTab, DialogTabs, ErrorTooltip, FormListItem, NeutralActionButton, ShortInput,
+		VisualSettingsButtonsContainer, VisualSettingsDialogContent,
+		NegativeActionButton } from './visualSettingsDialog.styles';
+import { isEqual } from 'lodash';
+import { DEFAULT_SETTINGS } from '../../../../../constants/viewer';
+
+const SettingsSchema = Yup.object().shape({
+	nearPlane: schema.number(0, Number.POSITIVE_INFINITY),
+	memory: schema.integer(16, 2032)
+});
+
+const BasicSettings = (props) => {
+	return (
+		<List>
+			<FormListItem>
+				Shading
+				<Field name="shading" render={ ({ field }) => (
+					<Select {...field}>
```

Comment:
Could you use `SelectField` instead of `Select`? `SelectField` pretend native behaviours like navigation between items using the keyboard. ;)

Reply:
Yest, that seems like a good idea